runtime/sema/account.cdc

```diff
@@ -205,6 +205,26 @@ struct Account {
         access(Contracts | UpdateContract)
         fun update(name: String, code: [UInt8]): DeployedContract
 
+        /// Updates the code for the contract/contract interface in the account,
+        /// and handle any deployment errors gracefully.
+        ///
+        /// The `code` parameter is the UTF-8 encoded representation of the source code.
+        /// The code must contain exactly one contract or contract interface,
+        /// which must have the same name as the `name` parameter.
+        ///
+        /// Does **not** run the initializer of the contract/contract interface again.
+        /// The contract instance in the world state stays as is.
+        ///
+        /// Fails if no contract/contract interface with the given name exists in the account,
+        /// if the given code does not declare exactly one contract or contract interface,
+        /// or if the given name does not match the name of the contract/contract interface declaration in the code.
+        ///
+        /// Returns the deployment result.
+        /// Result would contain the deployed contract for the updated contract, if the update was successfull.
+        /// Otherwise, the deployed contract would be nil.
+        access(Contracts | UpdateContract)
+        fun tryUpdate(name: String, code: [UInt8]): DeploymentResult
```

Comment:
Why not just have `DeployedContract?` be the return type? What is the reason for a separate container struct here?

Reply:
The plan is to also include some info about the error, so developers can decide what to do about it. Would eventually be something like:
```cadence
access(all) struct DeploymentResult {
    access(all) let deployedContract: DeployedContract?
    access(all) let deploymentError: DeploymentError?
}
```
(would have been better if it was something like `DeployedContract|DeploymentError` but we don't have union types)

But as discussed in https://github.com/onflow/cadence/issues/2700#issuecomment-1669992213, there is some work needed on the error/error-messages to make it happen, so it is being excluded for now.